### ISSUE TITLE
Integration with Apollo Federation

### ISSUE THREAD

**Issue by jacob-lester** [Urigo/graphql-modules#496]
I can't figure out how to initialize the ApolloServer schema using `buildFederatedSchema()` from `@apollo/federation`. (Read more about Apollo Federation [here](https://www.apollographql.com/docs/apollo-server/federation/introduction/))

My `AppModule` (`GraphQLModule`) has `typeDefs` and `resolvers` fields, but the resolvers type, `IResolvers` does not match the expected `GraphQLResolverMap` type parameter for resolvers.

```js
import 'reflect-metadata';
import { ApolloServer, gql } from 'apollo-server';
import { AppModule } from './modules/app.module';
import { buildFederatedSchema }from '@apollo/federation';
import './lib/dotenv.env';
import { environment } from './lib/apollo.env';

// Initialize Apollo GraphQL server using the entry AppModule
const server = new ApolloServer({
  // Also tried passing in AppModule.schema into buildFederatedSchema
  schema: buildFederatedSchema([{AppModule.typeDefs, AppModule.resolvers}]),
  context: session => session,
  tracing: environment.apollo.tracing,
  introspection: environment.apollo.introspection,
  playground: environment.apollo.playground
});

// Start the server on port 4000.
// Navigate to http://localhost:4000/graphql for GQL playground 
server.listen({ port: environment.port }).then(({ url }) => {
  console.log(`🚀  Server ready at ${url}`);
});
```

**Comment by jacob-lester** [comment]
I got the resolvers to work by passing in `AppModule.resolvers as any`. However, now I am getting `Unknown directive "key"` error when I try to implement federation. This comes after adding the `@key (fields: "id")` directive in my `.graphql` file.

**Comment by ardatan** [comment]
Fixed in 0.7.6!

**Comment by jacob-lester** [comment]
I still am getting the same error upon upgrading to 0.7.7
```
GraphQL-Modules Error: Schema is not valid!
   - #Module #C:\Users\...\src\modules\app.module.ts doesn't have a valid schema!
   -- Unknown directive "key".

   Possible solutions:
    - Check syntax errors in typeDefs
    - Make sure you import correct dependencies
```

Also, the example [apollo federation project](https://github.com/Urigo/graphql-modules/tree/master/examples/apollo-federation) seems to have the same issue.
